### PR TITLE
Convert to PDK - Add support for Puppet 8

### PR DESCRIPTION
**Pull request by anders-larsson**
Opened a pull request for this in our fork. Work done by @Phil-Friderici.

### CODE REVIEW

**Review by Phil-Friderici — APPROVED** [review]
Not exactly sure what to review here ?

File: manifests/mount.pp [hunk 1]
```diff
@@ -1,34 +1,73 @@
...
-
-  # validate params
-  validate_re($ensure, '^(present)|(unmounted)|(absent)|(mounted)$',
-    "types::mount::${name}::ensure is invalid and does not match the regex.")
-  validate_absolute_path($name)
-
   if $ensure != 'absent' {
     # ensure target exists
-    include ::common
+    include common
```
Comment (anders-larsson) [inline manifests/mount.pp:63]: Is this include really required? common::mkdir_p should be be accessible without the class being included in the catalogue.

File: manifests/package.pp [hunk 2]
```diff
@@ -1,32 +1,53 @@
...
-  $ensure            = 'present',
-  $adminfile         = undef,
-  $allowcdrom        = undef,
-  $configfiles       = undef,
-  $install_options   = undef,
-  $provider          = undef,
-  $responsefile      = undef,
-  $source            = undef,
-  $uninstall_options = undef,
+  Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'] $ensure            = 'present',
```
Comment (anders-larsson) [inline manifests/package.pp:41]: Any reason why not Stdlib::Ensure::Package is not used here?

```
Validate the value of the ensure parameter for a package

Alias of Variant[Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'], String[1]]```

**Review by anders-larsson — COMMENTED** [review]
Added some notes. Not sure if it is something we want?

File: manifests/mount.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/mount.pp:63]: Agreed, even if common is a passive class it shouldn't be needed to be included.

File: manifests/package.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/package.pp:41]: `Stdlib::Ensure::Package` isn't available in stdlib 5.0.0 that is the current minimum version for this module. Should I change it in our fork ?

Comment (anders-larsson) [reply manifests/package.pp:41]: Ah OK. Not sure if we should up the stdlib requirements or not.

Comment (anders-larsson) [reply manifests/package.pp:41]: Let's leave it for now.

**Comment by Phil-Friderici** [comment]
Let me merge this first before I create a new PR with your changes.

File: manifests/mount.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/mount.pp:63]: see #2

File: manifests/package.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/package.pp:41]: see #2

**Pull request merged**